packages/core/src/modules/connections/handlers/ConnectionRequestHandler.ts

```diff
@@ -32,16 +37,54 @@ export class ConnectionRequestHandler implements MessageHandler {
   }
 
   public async handle(messageContext: MessageHandlerInboundMessage<ConnectionRequestHandler>) {
-    const { connection, recipientKey, senderKey } = messageContext
+    const { connection, recipientKey, senderKey, message } = messageContext
 
     if (!recipientKey || !senderKey) {
       throw new AriesFrameworkError('Unable to process connection request without senderVerkey or recipientKey')
     }
 
-    const outOfBandRecord = await this.outOfBandService.findCreatedByRecipientKey(
-      messageContext.agentContext,
-      recipientKey
-    )
+    const parentThreadId = message.thread?.parentThreadId
+
+    const createOobRecord = async (did: string) => {
+      // If it's a request related to an implicit invitation, make sure destination DID is present in our wallet
+      const publicDid = tryParseDid(did)
+      if (
+        publicDid &&
+        !(await messageContext.agentContext.dependencyManager
+          .resolve(DidRepository)
+          .findCreatedDid(messageContext.agentContext, publicDid.did))
+      ) {
+        throw new AriesFrameworkError(`Referenced public did ${did} not found.`)
+      }
+
+      const outOfBandInvitation = new OutOfBandInvitation({
+        id: message.threadId,
+        label: '',
+        services: [did],
+        handshakeProtocols: [HandshakeProtocol.Connections],
+      })
+
+      const outOfBandRecord = new OutOfBandRecord({
+        role: OutOfBandRole.Sender,
+        state: OutOfBandState.AwaitResponse,
+        reusable: true,
+        autoAcceptConnection: this.connectionsModuleConfig.autoAcceptConnections,
```

Comment:
I see we take over autoAcceptConnections from the agent config. One thing we should be careful of here is that before you always had to take a manual step for people to be able to connect to you (receiveInvitation, or createInvitation). With implicit invitations it means that if someone gets hold of my did they'll be able to create connections with me.

I'm thinking whether we should add a new `autoAcceptImplicitInvitations` property that determines whether you auto accept these invitations. What do you think?

Reply:
Maybe we should even allow to create an out of band record for a specific did that specifies if the did can be connected to. But then we need creating oob invitations for public dids of course.

Just thinking how we can make it easy to connect, but also don't open the gates that allows every did in our wallet to be connected to (even peer:dids ?!?!?)

Reply:
Very good point regarding adding `autoAcceptImplicitInvitations`. 

I think we can add a flag somewhere in `DidCreateOptions` (and dids.import I think) to define whether this is a public did applicable to implicit invitations. Something like `isPublicDid` or `supportsImplicitInvitations` (always complicated to get a good naming...).

Reply:
Yes that makes sense 👍

We may be able to handle this is in the did resolver service so methods don't have to explicitly support it.

We should probably call it something related to didcomm though. Maybe `autoAcceptImplicitDidCommInvitations`, but that's getting too long... Or maybe we should never autoAccept for now, and require the user to handle the request (so there's at least still one manual step). Then later if we understand a bit better what the common use cases are we can add a config flag to automate it?

Reply:
I agree... let's make it manual acceptance for now and then we can automate it if we find it useful. The good thing about this manual approach is that when receiving a request, the event handler can tell what DID this relates to and it can decide based on that if it should accept it or simply ignore it.